torch/_inductor/comms.py

```diff
@@ -112,42 +113,52 @@ def reorder_communication_preserving_peak_memory(
     improvement = {snode: node_stats[snode].improvement for snode in node_stats}
     total_improvement = sum([improvement[snode] for snode in improvement])
     total_moves = sum([node_stats[snode].moves for snode in node_stats])
-    overlap_log.info(
-        "reorder_communication_preserving_peak_memory improved overlap by %f ns after %d reorders",
-        total_improvement,
-        total_moves,
-    )
 
+    reorder_log_str = (
+        f"reorder_communication_preserving_peak_memory improved overlap by {total_improvement} ns"
+        f" after {total_moves} reorders.\n"
+    )
+    headers = [
+        "Collective node",
+        "initial exposed",
+        "final exposed",
+        "improvement",
+        "limiting factor",
+        "moves",
+    ]
+    rows = [
+        [
+            node_summary(snode),
+            node_reorder_info.initial_exposed,
+            node_reorder_info.final_exposed,
+            node_reorder_info.improvement,
+            node_reorder_info.limiting_factor,
+            node_reorder_info.moves,
+        ]
+        for snode, node_reorder_info in node_stats.items()
+    ]
     if importlib.util.find_spec("tabulate"):
         from tabulate import tabulate
 
-        overlap_log.info(
-            tabulate(
-                [
-                    [
-                        node_summary(snode),
-                        node_reorder_info.initial_exposed,
-                        node_reorder_info.final_exposed,
-                        node_reorder_info.improvement,
-                        node_reorder_info.limiting_factor,
-                        node_reorder_info.moves,
-                    ]
-                    for snode, node_reorder_info in node_stats.items()
-                ],
-                headers=[
-                    "Collective node",
-                    "initial exposed",
-                    "final exposed",
-                    "improvement",
-                    "limiting factor",
-                    "moves",
-                ],
-            )
+        reorder_log_str += tabulate(
+            rows,
+            headers=headers,
         )
     else:
-        overlap_log.info(
-            "Please `pip install tabulate` to nicely render overlap stats."
+        reorder_log_str += (
+            "Please `pip install tabulate` to nicely render overlap stats.\n"
```

Comment:
side thought - I have no idea if internal jobs generally come with `tabulate` installed and/or if this is easy to change from the trainer side, but should the default be to print a slightly uglier string rather than to print nothing at all?

Reply:
the default is actually an ugly string. i think you just missed it (2 lines below).  I also tested that this works / shows up in tlparse properly.  And the output can even be copypasted into python if you want to reformat it with tabulate after the fact.